### ISSUE TITLE
VB closure not capturing me reference, causes runtime null reference

### ISSUE THREAD

**Issue by liqdfire** [dotnet/roslyn#4856]
### Known conditions required to create
*  Must be in constructor
*  Must reference class level variable
*  Must be inside some type of control or flow statement
    * If block, for each, switch case etc...
*  Must capture a variable inside the control or flow block
</br></br>

I walked through the roslyn code, it appears to me that because of the control / flow block and the closure variable captures, the compiler is creating a lambda frame. When the [LambdaRewriter](../blob/master/src/Compilers/VisualBasic/Portable/Lowering/LambdaRewriter/LambdaRewriter.vb) goes to introduce the frame into the tree, it sees the method as being a constructor, and sets the bound me expression for deferred initialization. However, when the bound calls are visited it never initializes the deferred bound me reference because it sees the top level method as a constructor; therefore, it does not get added to the syntax tree. When in the method IntroduceFrame if I change the code flow to initialize immediately instead of deferred it compiles correctly. 

##### Small snippet that repos the bug
``` vb
Imports System.Collections.Generic
Public Class Processor
    Private _units As New List(Of Integer)

    Public Sub New()
        Dim source = New List(Of Integer) From {0}
        If True Then
            Dim namedList1 = New List(Of Integer) From {1, 2}
            source.ForEach(Sub(listItem)
                               listItem += namedList1.Count
                               _units.Add(listItem)
                           End Sub)
        End If
    End Sub
End Class
```
</br>

##### Screen shot showing memory value of "ME" prior to entering the closure
![ME reference before closure entry](https://cloud.githubusercontent.com/assets/328512/9535600/1a9d24cc-4cf1-11e5-8352-60f0a4c69d5e.png)
</br>

##### Screen shot showing memory value of "ME" after entering the closure
![ME reference after closure entry](https://cloud.githubusercontent.com/assets/328512/9535997/12b7d87a-4cf5-11e5-8e9f-cd084b6b1788.png)
</br>

##### Reflector decompilation showing Closure field $VB$Me never being set
``` vb
Public Class Processor
    ' Methods
    Public Sub New()
        Dim list As New List(Of Integer) From { 0 }
        Dim e$__- As New _Closure$__1-0 With { _
            .$VB$Local_namedList1 = 1 _
        }
        list.ForEach(New Action(Of Integer)(AddressOf e$__-._Lambda$__0))
    End Sub


    ' Fields
    Private _units As List(Of Integer) = New List(Of Integer)

    ' Nested Types
    <CompilerGenerated> _
    Friend NotInheritable Class _Closure$__1-0
        ' Methods
        Friend Sub _Lambda$__0(ByVal listItem As Integer)
            listItem = (listItem + Me.$VB$Local_namedList1)
            Me.$VB$Me._units.Add(listItem)
        End Sub


        ' Fields
        Public $VB$Local_namedList1 As Integer
        Public $VB$Me As Processor
    End Class
End Class
```
</br>

##### Bound me expression being built in method IntroduceFrame, set as deferred initialization
![Bound Me Expression](https://cloud.githubusercontent.com/assets/328512/9536418/6f027550-4cf9-11e5-99b7-bde6269aaecf.png)

**Comment by AlekseyTs** [comment]
Looks like the same issue as #4899 and should be fixed by #4908.

**Comment by AlekseyTs** [comment]
@VSadov Please verify.
